util/util.go

```diff
@@ -203,3 +206,13 @@ func convHexToUint64(hexString string) (uint64, error) {
 	}
 	return uint64(result), nil
 }
+
+// BlockUntilSuccessfulFn is designed to wait until a specified number of Ethereum blocks have been
+// mined, periodically checking for the completion of a given function within each block interval.
+type BlockUntilSuccessfulFn func(ctx context.Context, c *ethclient.Client, f func() error) error
+
+func BlockUntilSuccessful(ctx context.Context, c *ethclient.Client, retryable func() error) error {
```

Comment:
hopefully this is a permanent home for this fxn. It has been moved around so many times hahaha

Reply:
Not sure, I don't really like `util/util.go` lol but I didn't want to refactor too much 🥶